src/main/java/com/google/devtools/build/lib/syntax/MethodLibrary.java

```diff
@@ -102,6 +105,51 @@ private static Object findExtreme(
     }
   }
 
+  private static Object evalKey(
+      Object obj, Object key, Location loc, Environment env, FuncallExpression ast)
+      throws EvalException, InterruptedException {
+    if (key == Runtime.NONE) {
+      return obj;
+    } else if (key instanceof BuiltinCallable) {
+      return ((BuiltinCallable) key).call(Arrays.asList(obj), Collections.emptyMap(), ast, env);
```

Comment:
Collections.singletonList would be more efficient than Arrays.asList since it does only one allocation (no array).